ktor-utils/common/src/io/ktor/util/debug/IdeDebugChecker.kt

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.debug
+
+internal expect object IntellijIdeaDebugDetector {
+    /**
+     * Checks whether Intellij Idea debugger is conected to the current Ktor server.
+     * May return true only for JVM debugger.
+     * */
+    val isDebuggerConnected: Boolean
```

Comment:
Note: in order to execute this code once the pipeline.execute is called, I have to make this common module.
But on the other hand:
a) idk if it's possible to easily check debugger connection on JS or Native
b) currently, there seems to be not much use of it for non-jvm